src/transformStateWithClones.js

```diff
@@ -7,7 +7,51 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+
+  const eachStateResult = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = addProperties(currentState, action.extraData);
```

Comment:
Ensure that the `action` object has an `extraData` property when the action type is 'addProperties'. If this property is missing, the function may not behave as expected.